Fix compatibility with Pyramid 1.8

fschulze opened the pull request:
This fixes #22, but there are 4 more test failures about derived_views which I don't understand yet.

mmerickel commented:
Previously those views were registered for both `IViewClassifier` and `IExceptionViewClassifier`. In Pyramid 1.8 they are registered only for `IExceptionViewClassifier` so probably the test should just be switched to that.

I noticed one other thing which is that https://github.com/Pylons/pyramid/pull/2757 dropped `pyramid.config.expand_action` because I had no idea anyone was using it but it's apparently quite heavily used throughout the tests. Since 1.8 this line will never work: https://github.com/Pylons/pyramid_zcml/blob/5d162f15213338b6aedaebca9bbfc7436d5c244f/pyramid_zcml/tests/test_units.py#L1353 No good recommendation here but I'd rather not bring it back into Pyramid...

fschulze commented:
I've switched to use ``IExceptionViewClassifier``. This also works with older Pyramid versions. I have [a branch](https://github.com/fschulze/pyramid_zcml/tree/old-pyramid-testing) where I added environments for testing that. Not sure you want that merged, because it isn't integrated with Travis and there is a PR open that changes Travis/tox stuff.

It seems like the fallback import ``from zope.configuration.config import expand_action`` works fine, so I don't see a need for action there.

Python 2.6 fails with Pyramid 1.8, but works with older Pyramid versions.

PyPy3 currently fails, because virtualenv or pip is incompatible with it, because it's currently only supporting Python 3.2.

stevepiercy commented:
FWIW Python 2.6 support was dropped in Pyramid 1.7, and we dropped PyPy3 support until it supports Python 3.4+. We dropped Python 3.2 in Pyramid 1.7, and Python 3.3 in Pyramid 1.8.

Pyramid 1.8 is tested and supported on Python 2.7, Python 3.4, Python 3.5, Python 3.6, and PyPy.

fschulze commented:
I have a branch now that tests everything with travis: https://github.com/fschulze/pyramid_zcml/tree/tox-travis

mmerickel commented:
Yep this all looks as expected. Thanks @fschulze. @mcdonc is the only one with access to pypi so he'll need to cut a new release or give someone access (you?) to do it.

mmerickel commented:
@fschulze Apparently I ended up with pypi ownership so I've given you pypi access if you wish to cut a release of pyramid_zcml.
